### DISCUSSION TITLE
How to set font size for WSL2 GUI menus

### DISCUSSION THREAD

**Discussion by snowgato** [microsoft/WSL#9446] [General] [answered]
### Version

Microsoft Windows [version 10.0.22623.746]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.10.102.1-microsoft-standard-WSL2

### Distro Version

Ubuntu 20.04

### Repro Steps

Hi,
I don't see how to enlarge or shrink font size for GUI menus.
Within text zone, Ctrl++ or Ctrk+mouse scrolling can enlarge or shrink text size, but I don't know how to change font size for WSL2 GUI menus and haven't found neither solutions googling.
Thanks for your help !

### Expected Behavior

Enlarge or Shrink font size for GUI menus

### Actual Behavior

font size for GUI menus is too small ...

**Comment by elsaco** [comment]
@snowgato you could use `gsettings` and adjust your fonts. Use `gsettings list-recusively | grep -i font` to see all the schemas that contain font settings. `org.gnome.desktop.wm.preferences` and `org.gnome.desktop.interface` are of importance. Type `gnome list-schemas` to show all of them.

Examples:

`gsettings get org.gnome.desktop.wm.preferences titlebar-font` to display the current titlebar font size

`gsettings set org.gnome.desktop.wm.preferences titlebar-font 'Cantarell Bold 14'` sets a new font size

**Comment by snowgato** [comment]
> @snowgato you could use `gsettings` and adjust your fonts. Use `gsettings list-recusively | grep -i font` to see all the schemas that contain font settings. `org.gnome.desktop.wm.preferences` and `org.gnome.desktop.interface` are of importance. Type `gnome list-schemas` to show all of them.
> 
> Examples:
> 
> `gsettings get org.gnome.desktop.wm.preferences titlebar-font` to display the current titlebar font size
> 
> `gsettings set org.gnome.desktop.wm.preferences titlebar-font 'Cantarell Bold 14'` sets a new font size
Hi @elsaco, Thanks for your answer
Actually, I'd like to set RStudio menu font size. This app doesnt rely on gnome desktop as it worked whereas I had not installed gnome desktop.
Anyway, I've installed gnome by doing: sudo apt update
sudo apt upgrade and sudo apt-install ubuntu-desktop gnome
and then did gsettings set org.gnome.desktop.wm.preferences titlebar-font 'Cantarell Bold 14' . Unfortunately, that doesn't change Rstudio menu font size .
I guess there might be some WSL tweak to set default GUI font size somewhere in WSL config files, but which ??

**Comment by elsaco** [comment]
In this case if `rstudio` doesn't offer an option to increase the font size for the interface you could use `fontconfig` and add your preferences to `fonts.conf`. However this sets font properties globally.

**Comment by elsaco** [comment, marked as answer]
Also, if you don't want to tweak the `fonts.conf` just use the `GDK_DPI_SCALE` option. Running `GDK_DPI_SCALE=2 rstudio` will make everything larger.

**Reply by snowgato** [reply]
I've looked to fontconfig tools and I've not found how to setup font size.
Nevertheless, launching rstudio with GDK_DPI_SCALE=2 rstudio worked perfectly.
As a matter of fact, GDK_DPI_SCALE=2 was a bit too much for me and GDK_DPI_SCALE=1.5 rstudio is great.
Thanks very much for your help

**Comment by Welding-Torch** [comment]
Hi, I'm trying to increase the font size for the WSL menus while running warp terminal. Running `GDK_DPI_SCALE=2` did not work for me.

Could this be because Warp uses Wayland?

**Reply by brendon-colburn** [reply]
[this link](https://github.com/microsoft/wslg/issues/590) fixed it for me but I had to do the config file in my user folder and put the script in place. Font is a bit jagged though
